### PR TITLE
[Layout Legend] Children symbol node obey spacing according to layer node mode.

### LINKED ISSUE #50780: [Layout] Legend Item Indentation settings are applied on a layer level and not a symbol level
### What is the bug or the crash?

The `Indent of group items` and `Indent of subgroup items` are not properly applied in the legend, only at layer scale.

The indentation parameters are applies to the layers of a legend group (wether it'S defined as a sub-group or group), though if the layer (the root node) is defined as a group or a subgroup, the indentation will not be applied to it's children.

The needed workaround is to generate a group to host the layer with multipe symbols in order to differenciate it from layers with a single symbol level.

### Steps to reproduce the issue

1. Create a map
2. Set the symbology of a vector layer to have multiple symbols (rule-based, categorized, range,etc)
3. Create a layout
4. Set the layer to be a group or subgroup
5.  In spacing, set the indent value for the group or subgroup level
6. Nothing is indented

### Versions

Tested on 3.27 and 3.22.11 on win 10

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by roya0045**
## Description

Fixes #50780

This PR proposed to have layers with multiple children symbol node act as a group or subgroup and use the value set in the spacing settings. 

This way a user doesn't need to create a roup in the legend to have a the symbol node of a single layer indented. As the layer name currently obey the font, it makes sens to also have it follow the spacing.
image

### CODE REVIEW

**Comment by roya0045** [comment]
Failed test is test_core_rasterfilewriter, unrelated (finally).

File: src/core/qgslegendrenderer.cpp [hunk 1]
```diff
@@ -379,7 +379,15 @@ QList<QgsLegendRenderer::LegendComponentGroup> QgsLegendRenderer::createComponen
...
-
+      double symbolIndent = indent;
+      if ( layerStyle == QgsLegendStyle::Subgroup )
+      {
+        symbolIndent += mSettings.style( QgsLegendStyle::Subgroup ).indent( );
+      }
+      else if ( layerStyle == QgsLegendStyle::Group )
+      {
+        symbolIndent += mSettings.style( QgsLegendStyle::Group ).indent( );
+      }
```
Comment (nyalldawson) [inline src/core/qgslegendrenderer.cpp:390]:
```suggestion
     switch( layerStyle )
     {
        case QgsLegendStyle::Subgroup:
        case QgsLegendStyle::Group:
            symbolIndent += mSettings.style( layerStyle ).indent( );
            break;
         case ...:
            break;
      }
```

**Comment by nyalldawson** [comment]
looking good to me!

**Review by nyalldawson — APPROVED** [review]

**Comment by roya0045** [comment]
Thanks Nyall for marging this

**Pull request merged**